### PR TITLE
Ports TG Pickup, Equip, and Drop Sounds

### PR DESCRIPTION
**Pull request by Fox-McCloud**
Ports TG's equip, pickup, and drop sounds, credit to them for this, and especially @Qustinnus 

Adds a system for items having a pickup, drop, and equip sound---helps the game seem more lively and immersive.

Demonstration video, from TG, courtesy of @Qustinnus:
https://streamable.com/koz4r

Currently, belts and uniforms are the items that have specific equip sounds.

Code wise, this should have been straight forward, but our inventory code is very dated and a royal pain; this should work well, day to day, but there will be some instances where a drop sound plays where it ideally shouldn't (mostly when dropping things into a disposal bin or adding an item to something like an autolathe), that said, for all other inventory reactions, there shouldn't be an overlap of sounds.

Also adds tape recorder sounds, as well

Since this had to touch all over item throwing and mobs getting hit by items, I just went ahead and ported TG's "if a hot item is thrown at you and hits you and you're flammable, you ignite".

:cl: Fox McCloud
add: Adds in a bunch of item pick up, drop, and equipping sounds
add: hitting flammable mobs with a thrown item that is hot will ignite them
add: tape recorders have a whole new set of sounds when they're operating
/:cl:

### CODE REVIEW

File: code/__DEFINES/flags.dm [hunk 1]
```diff
@@ -114,22 +114,21 @@
...
-#define SLOT_EYES 8
-#define SLOT_EARS 16
-#define SLOT_MASK 32
-#define SLOT_HEAD 64
-#define SLOT_FEET 128
-#define SLOT_ID 256
-#define SLOT_BELT 512
-#define SLOT_BACK 1024
-#define SLOT_POCKET 2048		//this is to allow items with a w_class of 3 or 4 to fit in pockets.
-#define SLOT_DENYPOCKET 4096	//this is to deny items with a w_class of 2 or 1 to fit in pockets.
```
Comment (Fox-McCloud) [inline code/__DEFINES/flags.dm:129]: I killed this off because this is unused.

File: code/__HELPERS/unsorted.dm [hunk 2]
```diff
@@ -2079,3 +2079,30 @@ GLOBAL_DATUM_INIT(dview_mob, /mob/dview, new)
 	))
 	query_accesslog.warn_execute()
 	qdel(query_accesslog)
+
+/proc/slot_bitfield_to_slot(input_slot_flags) // Kill off this garbage ASAP; slot flags and clothing flags should be IDENTICAL. GOSH DARN IT. Doesn't work with ears or pockets, either.
```
Comment (Fox-McCloud) [inline code/__HELPERS/unsorted.dm:2083]: Awful, just awful. 

we define slots with `slot_x` that are just numerical numbers....ideally these would all just be the same thing, but here we are---this is definitely one of those "please do this refactor, pleaseeee" situations.

**Comment by SabreML** [comment]
Found a couple of issues:
* There's overlapping sounds when you pick up a storage container (Boxes, toolboxes, bibles, etc.)
* Also overlaps if you unequip or pick up a toolbelt
* There doesn't seem to be any sound for putting things *into* a toolbelt, although taking them out works just fine.
* And of course the other stuff you mentioned in the description is there too.

**Comment by Fox-McCloud** [comment]
>There's overlapping sounds when you pick up a storage container (Boxes, toolboxes, bibles, etc.)

Also how it functions on TG; I presumed this was intentional.

>There doesn't seem to be any sound for putting things into a toolbelt, although taking them out works just fine.

This has never been the case.

**Comment by Kiyahitayika** [comment]
Love the work that's been going on for this server! @Onyxarias has been talking to me about /tg/ sounds for months!

**Comment by Qustinnus** [comment]
just wait until fox ports sound environments ;)

**Review by Kyep — APPROVED** [review]

**Comment by Fox-McCloud** [comment]
@Qustinnus 

Already have! It's in!

**Comment by Onyxarias** [comment]
Yeahhh, I was looking at adding these for MONTHS. I'm happy a much more talented coder got around to it, before I inevitably tried and did a poor job of it :p

File: code/__HELPERS/unsorted.dm (hunk 2 above)
Comment (AffectedArc07) [reply code/__HELPERS/unsorted.dm:2083]: The real sin here is use of lowercase defines
```dm
// slots
#define slot_back 1
#define slot_wear_mask 2
#define slot_handcuffed 3
#define slot_l_hand 4
#define slot_r_hand 5
#define slot_belt 6
#define slot_wear_id 7
#define slot_l_ear 8
#define slot_glasses 9
#define slot_gloves 10
#define slot_head 11
#define slot_shoes 12
```

**Review by AffectedArc07 — APPROVED** [review]

**Pull request merged**